pkg/services/accesscontrol/resourcepermissions/types/hook.go

```diff
@@ -1,7 +1,14 @@
 package types
 
-import "github.com/grafana/grafana/pkg/services/sqlstore"
+import (
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
 
-type UserResourceHookFunc func(session *sqlstore.DBSession, orgID, userID int64, resourceID, permission string) error
+type UserResourceHookFunc func(session *sqlstore.DBSession, orgID int64, user User, resourceID, permission string) error
 type TeamResourceHookFunc func(session *sqlstore.DBSession, orgID, teamID int64, resourceID, permission string) error
 type BuiltinResourceHookFunc func(session *sqlstore.DBSession, orgID int64, builtInRole, resourceID, permission string) error
+
+type User struct {
```

Comment:
nit: I think we should move this from the internal types package to the access control package or just reuse , WDYT?

Reply:
Good point, I also wasn't sure it was the right place. I've moved it to `accesscontrol` package now.